src/core/main.c

```diff
@@ -3181,7 +3181,17 @@ int main(int argc, char *argv[]) {
 #endif
 
 #if HAS_FEATURE_ADDRESS_SANITIZER
+        /* At this stage we most likely don't have stdio/stderr open, so the following
+         * LSan check would not print any actionable information and would just crash
+         * PID 1. To make this a bit more helpful, let's try to open /dev/console,
+         * and if we succeed redirect LSan's report there. */
+
+        int tty_fd = open_terminal("/dev/console", O_WRONLY|O_NOCTTY|O_CLOEXEC);
+        if (tty_fd >= 0)
+            __sanitizer_set_report_fd((void*) (intptr_t) tty_fd);
```

Comment:
indentation... 8ch please

Reply:
this should be conditioned to `getpid_cached() == 1` really.

Reply:
(i.e. the --user instance shouldn't really try that)

Reply:
(Ouch. I reviewed the PR from my phone, and the shown layout is broken. So, I did not notice that.)

Reply:
Ugh, sorry about that. I really need to figure out what keeps stealing my whitespaces... I'll do a follow up once I'm back at my pc.

Reply:
-> https://github.com/systemd/systemd/pull/29206